scripts/start.py

```diff
@@ -78,146 +73,131 @@
     action='store_true')
 _PARSER.add_argument(
     '--no_auto_restart',
-    help=(
-        'optional; if specified, does not automatically restart when files are '
-        'changed.'),
+    help='optional; if specified, does not automatically restart when files '
+         'are changed.',
     action='store_true')
 _PARSER.add_argument(
     '--source_maps',
-    help=(
-        'optional; if specified, build webpack with source maps.'),
+    help='optional; if specified, build webpack with source maps.',
     action='store_true')
 
 PORT_NUMBER_FOR_GAE_SERVER = 8181
 
 
-def cleanup():
-    """Wait for the servers to go down and set constants back to default
-    values.
+@contextlib.contextmanager
+def alert_on_exit():
+    """Context manager that alerts developers to wait for a graceful shutdown.
+
+    Yields:
+        None. Nothing.
     """
-    common.print_each_string_after_two_new_lines([
-        'INFORMATION',
-        'Cleaning up the servers.'])
-    while common.is_port_in_use(PORT_NUMBER_FOR_GAE_SERVER):
-        time.sleep(1)
-    build.set_constants_to_default()
-    common.stop_redis_server()
+    try:
+        yield
+    finally:
+        python_utils.PRINT(
+            '\n\n'
+            # ANSI escape sequence for bright yellow text color.
+            '\033[93m'
+            # ANSI escape sequence for bold font.
+            '\033[1m'
+            'Servers are shutting down, please wait for them to end gracefully!'
+            # ANSI escape sequence for resetting formatting.
+            '\033[0m'
+            '\n\n')
+        # Give developers an opportunity to read the alert.
+        time.sleep(5)
+
+
+def notify_successful_shutdown():
```

Comment:
notify is transitive, so perhaps notify_about_successful_shutdown

Reply:
Done.